addons/stock/models/stock_move.py

```diff
@@ -400,7 +400,10 @@ def write(self, vals):
                     lambda m: m.state not in ['draft', 'done', 'cancel'] and float_compare(m.reserved_availability, vals.get('product_uom_qty'), precision_rounding=m.product_uom.rounding) == 1
                 )
                 move_to_unreserve._do_unreserve()
-                (self - move_to_unreserve).filtered(lambda m: m.state == 'assigned').write({'state': 'partially_available'})
+                assigned_moves = (self - move_to_unreserve).filtered(lambda m: m.state == 'assigned')
+                move_partial_available = assigned_moves.filtered((lambda m: m.reserved_availability > 0))
```

Comment:
Use `float_is_zero` instead. (also there is a extra parentheses in your filtered)